Customizer service and enhancements

jamesremuscat opened the pull request:
This PR moves `cluster-customizer` actions out of the `clusterware-handlers` repository, as well as making several enhancements to the customizer:

- The concept of customization repositories is introduced; see `clusterware-services` PR for more details.
- The master node now serves a configuration file over HTTP (via `alces-flight-www`) which is loaded by compute nodes, from which compute nodes configure customization repositories and initial profiles to apply. (This is in addition to profiles specified in user data, a mechanism which remains unchanged.)
- Account profiles are now consistently referred to as such, rather than the generic "profiles".

Related PRs:
 - `clusterware-services`: https://github.com/alces-software/clusterware-services/pull/37
- `flight-profiles`: TBC

jamesremuscat committed:
Move customizer lib/libexec into clusterware-services.

jamesremuscat committed:
Use `alces-flight-www` to serve up a config file for compute nodes.

jamesremuscat committed:
Load customizer config file from master if we know its IP.

jamesremuscat committed:
We need to extract master IP from the YAML ourselves.

jamesremuscat committed:
Tidy indentation.

jamesremuscat committed:
Wait for cloud-init to have written the `config.yml` file.

jamesremuscat committed:
Derp. Use correct variable for master IP.

jamesremuscat committed:
Include default repositories.

jamesremuscat committed:
repositories should be a dictionary not an array.

jamesremuscat committed:
Default account bucket now referred to as 'account' not 'profile'.

jamesremuscat committed:
Use the `customize-repository` system to apply some profiles at init-time.

We now use the new repo system to apply profiles specified in the config file
obtained from the master node, while retaining the existing fetch system for
profiles specified in user data to cloud-init.

jamesremuscat committed:
Rename mention of "profiles" to "account profiles".

mjtko commented on cluster-customizer/initialize:
```diff
@@ -166,12 +222,13 @@ setup
 files_load_config cluster-customizer
 cw_CLUSTER_CUSTOMIZER_path="${cw_CLUSTER_CUSTOMIZER_path:-${cw_ROOT}/var/lib/customizer}"
-cw_CLUSTER_CUSTOMIZER_profiles="${cw_CLUSTER_CUSTOMIZER_profiles:-default}"
+cw_CLUSTER_CUSTOMIZER_account_profiles="${cw_CLUSTER_CUSTOMIZER_account_profiles:-default}"
```
Could we add a little backwards compatibility in here and do something like this please?

```
${cw_CLUSTER_CUSTOMIZER_account_profiles:-${cw_CLUSTER_CUSTOMIZER_profiles:-default}}
```

mjtko replied:
(Mainly cos I'm worrying about things that already make use of the older name.)

jamesremuscat committed:
Backwards-compatibility: fall back to old variable name first.

This maintains compatibility with anything that still writes the old
`cw_CLUSTER_CUSTOMIZER_profiles` variable and expects it to be read.
